### PR TITLE
Changelog for 0.29.0

### PR DESCRIPTION
**Pull request by angelozerr**
Changelog for 0.29.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
 # Change Log
 
+## [0.29.0](https://github.com/eclipse/lemminx/milestone/45?closed=1) (November 21, 2024)
+
+### Enhancements
+
+- ICompletionParticipant#onXMLContent is not called for CDATA. See [#1694](https://github.com/eclipse-lemminx/lemminx/issues/1694).
+
+### Bug Fixes
```
Comment (datho7561) [inline CHANGELOG.md:9]: Is https://github.com/eclipse-lemminx/lemminx/commit/d3a123531c8b48d6139759beab6ea1fdaeefe158 worth adding?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
 # Change Log
 
+## [0.29.0](https://github.com/eclipse/lemminx/milestone/45?closed=1) (November 21, 2024)
+
+### Enhancements
+
+- ICompletionParticipant#onXMLContent is not called for CDATA. See [#1694](https://github.com/eclipse-lemminx/lemminx/issues/1694).
```
Comment (datho7561) [inline CHANGELOG.md:7]: I think this is a bug fix. i.e. we weren't considering completion inside CDATA as XML content, but now we are.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+
+### Bug Fixes
+
+- Supports TextDocumentContentChangeEvent#getRangeLength with null value. See [#1696](https://github.com/eclipse-lemminx/lemminx/pull/1696).
+- org.eclipse.lemminx.dom.DOMNode.getTextContent() seem to be implemented wrong (or incomplete). See [#1695](https://github.com/eclipse-lemminx/lemminx/issues/1695).
```
Comment (datho7561) [inline CHANGELOG.md:12]: maybe reword to "Implement `org.eclipse.lemminx.dom.DOMNode.getTextContent()` according to the API", similar to how the PR is named

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (angelozerr) [reply CHANGELOG.md:6]: Good catch. Added.

File: CHANGELOG.md (hunk 2 above)
Comment (angelozerr) [reply CHANGELOG.md]: move to bug

File: CHANGELOG.md (hunk 3 above)
Comment (angelozerr) [reply CHANGELOG.md]: renamed

**Comment by laeubi** [comment]
As it seems the release has not happened yet, can we probably consider this one for the release as well?

- https://github.com/eclipse-lemminx/lemminx/pull/1710

it is mostly adding some headers to the manifest so no real code changes.

**Comment by laeubi** [comment]
Tanks you very much!

**Comment by angelozerr** [comment]
> Tanks you very much!

You are welcome ! @datho7561 will try to create the release today.

**Comment by datho7561** [comment]
@laeubi I just finished performing the release for lemminx
